### PR TITLE
AK+Everywhere: Add ReadonlySpan<T> alias and use it everywhere 

### PR DESCRIPTION
**Pull request by MacDue**
This adds `ReadonlySpan<T>` as an alias for `Span<T const>`, this was something brought up in #17309. 

With `Span<T const>` it's a little less clear that means underlying array is read-only, and the syntax is a little more awkward (and can easily be forgotten, as it has in places before). 

(Almost no functional changes, one tiny removal of some unnecessary mutation)

### CODE REVIEW

File: Ladybird/WebContentView.cpp [hunk 1]
```diff
@@ -576,7 +576,7 @@ void WebContentView::create_client()
 
         auto webcontent_fd_passing_socket_string = DeprecatedString::number(wc_fd_passing_fd);
 
-        Vector<StringView> arguments {
+        Vector<StringView, 5> arguments {
```
Comment (trflynn89) [inline Ladybird/WebContentView.cpp:579]: Why?

File: AK/Span.h [hunk 2]
```diff
@@ -271,13 +271,17 @@ struct Traits<Span<T>> : public GenericTraits<Span<T>> {
     constexpr static bool is_trivial() { return true; }
 };
 
-using ReadonlyBytes = Span<u8 const>;
+template<typename T>
+using ReadonlySpan = Span<T const>;
+
+using ReadonlyBytes = ReadonlySpan<u8>;
```
Comment (trflynn89) [inline AK/Span.h:277]: I'm pretty surprised we don't get redefinition errors on these :thinking:

File: Ladybird/WebContentView.cpp (hunk 1 above)
Comment (MacDue) [reply Ladybird/WebContentView.cpp:579]: There's a maximum of 5 arguments here, so I thought making it a little stack array would be nice (it's not an important thing).

Comment (trflynn89) [reply Ladybird/WebContentView.cpp:579]: That's a bit textbook premature IMO, but anyways, in the future a change like that should be in its own commit. It had nothing to due with the stated commit message / PR as a whole.

**Pull request merged**